bftengine/cmf/execution_data.cmf

```diff
@@ -0,0 +1,22 @@
+
+Msg RequestInput 1 {
+    uint16 clientId
+    uint64 executionSequenceNum
+    string cid
+    uint64 flags
+    string request
+    string signature
+    uint64 requestSequenceNum
+}
+
+Msg Timestamp 2 {
+    int64 time_since_epoch
+    uint64 request_position
```

Comment:
What is request_position in the context of a timestamp? Btw, CMF allows comments :)

Reply:
The new time may add a small amount of time to the given ```time_since_epoch``` per request i.e. if a request is the 3rd in the queue the command handler will calculate its execution time as ```time_since_epoch``` + 3*epsilon.

If the execution mode is not accumulated i.e. execution of a single request per execute, the command handler needs to know the ```request_position``` of the request in the list.   
@f-squirrel

Reply:
That makes sense to me. However, the message organization doesn't make sense to me yet.

The `ExecutedREquests` message contains a list of requests and a single timestamp with a `request_position`. How does the single request_position relate to the list of requests? This might just be a terminology issue.

Reply:
@robem , `request_position` represents the position of `ExecutedREquests` in the PrePrepare.
I agree that the name does not fully explain the purpose of the field but I could not find a better one.

Reply:
I will add a comment as it's not a trivial field